### PR TITLE
util: verify_access_key

### PR DESCRIPTION
**Pull request by WilliamEspegren**
Verify that the user provided a valid openai token by creating a request to the openAI endpoint.

### CODE REVIEW

File: litellm/utils.py [hunk 1]
```diff
@@ -2554,3 +2554,18 @@ def trim_messages(
     except: # [NON-Blocking, if error occurs just return final_messages
         return messages
 
+
+# Verify that the user has passed in a valid and active api key
+def verify_access_key(access_key:str):
+    openai.api_key = access_key
+    try:
+        test_response = openai.ChatCompletion.create(
+            model="gpt-3.5-turbo",
```
Comment (ishaan-jaff) [inline litellm/utils.py:2613]: can we do this for all litellm supported llms ? you can use `litellm.completion`

File: litellm/utils.py [hunk 2]
```diff
@@ -2554,3 +2554,18 @@ def trim_messages(
     except: # [NON-Blocking, if error occurs just return final_messages
         return messages
 
+
+# Verify that the user has passed in a valid and active api key
+def verify_access_key(access_key:str):
```
Comment (ishaan-jaff) [inline litellm/utils.py:2609]: can we modify this to accept key, model

File: litellm/tests/test_verify_openai_key.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from litellm import verify_access_key
+
+def bad_key():
+    key = "bad-key"
+    response = verify_access_key(key)
+    print(f"response: {response}")
+bad_key()
```
Comment (ishaan-jaff) [inline litellm/tests/test_verify_openai_key.py]: can you add a test for good key. you can use os.environ['OPENAI_API_KEY'] for the good key. Our CI/CD pipeline uses that too

File: litellm/utils.py (hunk 2 above)
Comment (WilliamEspegren) [reply litellm/utils.py:2609]: Should we check if the model they named is a valid model from openAI?

File: litellm/tests/test_verify_openai_key.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from litellm import verify_access_key
+
+def bad_key():
```
Comment (ishaan-jaff) [inline litellm/tests/test_verify_openai_key.py]: rename this to test_bad_key

File: litellm/utils.py (hunk 1 above)
Comment (WilliamEspegren) [reply litellm/utils.py:2613]: What do you mean? We only need to check if the token is valid, hence we use one of the cheapest models?

Comment (ishaan-jaff) [reply litellm/utils.py:2613]: but users should be able to check if their anthropic, replicate, huggingface key etc is valid too

File: litellm/utils.py (hunk 2 above)
Comment (WilliamEspegren) [reply litellm/utils.py:2609]: We do this in another PR

File: litellm/utils.py (hunk 1 above)
Comment (WilliamEspegren) [reply litellm/utils.py:2613]: Will do it soon 👍

**Comment by ghost** [comment]
If i'm not mistaken, isn't this the same as this? 
Screenshot 2023-09-12 at 11 33 55 AM


I guess we're missing docs on this 

@WilliamEspegren @ishaan-jaff

**Comment by ishaan-jaff** [comment]
yes it's the same! thanks for pointing this out @krrishdholakia 

@WilliamEspegren wrote good tests on this + is passing max_tokens = 10

**Comment by ishaan-jaff** [comment]
lets stick to using check_valid_key on litellm

I'll merge this PR to add tests written by @WilliamEspegren

**Comment by WilliamEspegren** [comment]
Perfect! Teamwork makes the dreamwork 🤝

**Comment by ghost** [comment]
Great work @WilliamEspegren 

We'll announce this in our changelog today - if you have a twitter/linkedin you'd like us to shoutout, can you share that here?

**Comment by WilliamEspegren** [comment]
Yes please, thank you. Here is my LinkedIn: https://www.linkedin.com/in/william-espegren/

**Comment by ishaan-jaff** [comment]
@WilliamEspegren you can use `litellm.utils.check_valid_key` in your open-interpreter PR 

will lyk what litellm version number once deployed

**Comment by WilliamEspegren** [comment]
Exactly 🔥, thank you 🙏

**Comment by WilliamEspegren** [comment]
I have tested inputting "sk-dsadasdasdadsadasdsadasdasdasdasdaasdasadaasdasa" as the api_key and I get the last (ValueError thrown) which isn't really what we are looking for. If we replace "except AuthenticationError as e:" with "except openai.error.AuthenticationError as e:" it works fine. But then this is only for openAI models. We might can skip this whole step and just throw that the key is invalid if it fails to create the completion?

**Comment by ishaan-jaff** [comment]
@WilliamEspegren I updated this, it's fixed in the latest version (just waiting for a deploy)

**Comment by ishaan-jaff** [comment]
Can you try litellm 0.1.605 @WilliamEspegren

**Comment by WilliamEspegren** [comment]
@ishaan-jaff Yes, I can try it 👍

**Comment by WilliamEspegren** [comment]
Now it works like a charm 🤜🤛

**Comment by ishaan-jaff** [comment]
🤜🤛

**Pull request merged**